packages/components/src/Select/Select.rebuilt.test.tsx

```diff
@@ -0,0 +1,258 @@
+        </SelectRebuilt>,
+      );
+
+      const select = container.querySelector("select") as HTMLSelectElement;
+      fireEvent.focus(select);
+
+      expect(focusHandler).toHaveBeenCalledTimes(1);
+    });
+
+    it("should call onBlur when the select loses focus", () => {
+      const blurHandler = jest.fn();
+
+      const { container } = render(
+        <SelectRebuilt version={2} onBlur={blurHandler}>
+          <Option value="foo">Foo</Option>
+          <Option value="bar">Bar</Option>
+        </SelectRebuilt>,
+      );
+
+      const select = container.querySelector("select") as HTMLSelectElement;
+      fireEvent.blur(select);
+
+      expect(blurHandler).toHaveBeenCalledTimes(1);
+    });
+
+    it("should handle all events correctly when they are combined", () => {
+      const changeHandler = jest.fn();
+      const focusHandler = jest.fn();
+      const blurHandler = jest.fn();
+
+      const { container } = render(
+        <SelectRebuilt
+          version={2}
+          onChange={changeHandler}
+          onFocus={focusHandler}
+          onBlur={blurHandler}
+          value="foo"
+        >
+          <Option value="foo">Foo</Option>
+          <Option value="bar">Bar</Option>
+          <Option value="baz">Baz</Option>
+        </SelectRebuilt>,
+      );
+
+      const select = container.querySelector("select") as HTMLSelectElement;
+
+      // Test focus
+      fireEvent.focus(select);
+      expect(focusHandler).toHaveBeenCalledTimes(1);
+
+      // Test change
+      fireEvent.change(select, { target: { value: "bar" } });
+      expect(changeHandler).toHaveBeenCalledTimes(1);
+      expect(changeHandler).toHaveBeenCalledWith("bar", expect.any(Object));
+
+      // Test blur
+      fireEvent.blur(select);
+      expect(blurHandler).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("inputRef", () => {
+    it("forwards inputRef to the select element", () => {
+      const ref = React.createRef<HTMLSelectElement>();
+
+      render(
+        <SelectRebuilt version={2} inputRef={ref}>
+          <Option>Foo</Option>
+        </SelectRebuilt>,
+      );
+
+      expect(ref.current).toBeInstanceOf(HTMLSelectElement);
+    });
+
+    it("can access native select methods through inputRef", () => {
+      const ref = React.createRef<HTMLSelectElement>();
+
+      const { container } = render(
+        <SelectRebuilt version={2} inputRef={ref}>
+          <Option>Foo</Option>
+        </SelectRebuilt>,
+      );
+
+      const selectElement = container.querySelector("select");
+
+      // Setup mocks
+      if (selectElement) {
+        selectElement.focus = jest.fn();
+        selectElement.blur = jest.fn();
+      }
+
+      // Call native methods through ref
+      if (ref.current) {
+        ref.current.focus();
+        ref.current.blur();
+      }
+
+      // Verify
+      expect(selectElement?.focus).toHaveBeenCalled();
+      expect(selectElement?.blur).toHaveBeenCalled();
+    });
```

Comment:
You can use typescript expects to not have to do the conditional checks and other changes as well.

```suggestion
    it("can access native select methods through inputRef", () => {
      const ref = React.createRef<HTMLSelectElement>();
      const mockFocus = jest.fn();
      const mockBlur = jest.fn();
      render(
        <SelectRebuilt version={2} inputRef={ref}>
          <Option>Foo</Option>
        </SelectRebuilt>,
      );

      const selectElement = screen.getByRole("combobox");
      expect(selectElement).toBeInstanceOf(HTMLSelectElement);
      // Setup mocks
      selectElement.focus = mockFocus;
      selectElement.blur = mockBlur;

      // Call native methods through ref
      ref.current?.focus();
      ref.current?.blur();

      // Verify
      expect(mockFocus).toHaveBeenCalled();
      expect(mockBlur).toHaveBeenCalled();
    });
```